numpy/core/src/multiarray/einsum.c.src

```diff
@@ -2745,19 +2745,20 @@ PyArray_EinsteinSum(char *subscripts, npy_intp nop,
                     NPY_ITER_ALIGNED|
                     NPY_ITER_ALLOCATE|
                     NPY_ITER_NO_BROADCAST;
-
+    iter_flags = NPY_ITER_EXTERNAL_LOOP|
+            ((dtype != NULL) ? 0 : NPY_ITER_COMMON_DTYPE)|
+            NPY_ITER_BUFFERED|
+            NPY_ITER_DELAY_BUFALLOC|
+            NPY_ITER_GROWINNER|
+            NPY_ITER_REDUCE_OK|
+            NPY_ITER_REFS_OK|
+            NPY_ITER_ZEROSIZE_OK;
+    if (out != NULL) {
+        iter_flags |= NPY_ITER_COPY_IF_OVERLAP;
+    }
```

Comment:
Would be clearer to move `((dtype != NULL) ? 0 : NPY_ITER_COMMON_DTYPE)` out of the ternary and into a similar if

Reply:
I think once that minor change has been made, this is ready to go in.

Reply:
changed